File: sentences/precompute/parity_mpnet.py
import json, math
from sentence_transformers import SentenceTransformer
s = json.load(open('parity_sentences.json', encoding='utf-8'))
m = SentenceTransformer('sentence-transformers/all-mpnet-base-v2')
py = m.encode(s, normalize_embeddings=True, convert_to_numpy=True)
js = json.load(open('parity_mpnet_js.json'))
def cos(a,b):
    d=sum(x*y for x,y in zip(a,b)); na=math.sqrt(sum(x*x for x in a)); nb=math.sqrt(sum(y*y for y in b))
    return d/(na*nb) if na and nb else 0
worst=min(cos(list(map(float,py[i])), js[i]) for i in range(len(js)))
print(f'mpnet parity worst cosine: {worst:.6f}  ->', 'PASS' if worst>0.999 else 'FAIL')

File: sentences/precompute/parity_mpnet.mjs
import { pipeline } from '@huggingface/transformers';
import { readFileSync, writeFileSync } from 'node:fs';
const s = JSON.parse(readFileSync('parity_sentences.json','utf8'));
const ex = await pipeline('feature-extraction', 'Xenova/all-mpnet-base-v2');
const out = [];
for (const t of s) { const o = await ex(t,{pooling:'mean',normalize:true}); out.push(Array.from(o.data, x=>Math.round(x*1e6)/1e6)); }
writeFileSync('parity_mpnet_js.json', JSON.stringify(out));
console.log(`mpnet js: ${out.length} x ${out[0].length}`);
